app/src/main/java/com/space/quiz_app/presentation/quiz_home_screen/view_model/QuizHomeViewModel.kt

```diff
@@ -0,0 +1,90 @@
+package com.space.quiz_app.presentation.quiz_home_screen.view_model
+
+import androidx.navigation.fragment.NavHostFragment
+import com.space.quiz_app.R
+import com.space.quiz_app.common.extensions.viewModelScope
+import com.space.quiz_app.data.remote.service.result_handler.resource.Resource
+import com.space.quiz_app.domain.model.user.QuizUserDomainModel
+import com.space.quiz_app.domain.repository.QuizSubjectsRepository
+import com.space.quiz_app.domain.repository.QuizUserRepository
+import com.space.quiz_app.presentation.base.view_model.QuizBaseViewModel
+import com.space.quiz_app.presentation.mapper.question.QuizQuestionDomainMapper
+import com.space.quiz_app.presentation.mapper.user.QuizUserDomainToUIMapper
+import com.space.quiz_app.presentation.mapper.user.QuizUserUIToDomainMapper
+import com.space.quiz_app.presentation.model.questions.QuizQuestionsUIModel
+import com.space.quiz_app.presentation.model.user.QuizUserUIModel
+import com.space.quiz_app.presentation.quiz_home_screen.ui.QuizHomeFragment
+import com.space.quiz_app.presentation.quiz_home_screen.ui.QuizHomeFragmentDirections
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class QuizHomeViewModel(
+    private val quizUserRepository: QuizUserRepository,
+    private val quizSubjectsRepository: QuizSubjectsRepository,
+    private val quizQuestionsUIMapper: QuizQuestionDomainMapper,
+    private val quizUserUIToDomainMapper: QuizUserUIToDomainMapper,
+    private val quizUserDomainToUIMapper: QuizUserDomainToUIMapper
+) : QuizBaseViewModel() {
+
+    private val _usernameState = MutableStateFlow("")
+    val usernameState = _usernameState.asStateFlow()
+
+    private val _subjectsState =
+        MutableStateFlow<List<QuizQuestionsUIModel>?>(null)
+    val subjectsState = _subjectsState.asStateFlow()
+
+    private val _errorState =
+        MutableStateFlow<Throwable?>(null)
+    val errorState = _errorState.asStateFlow()
+
+    private val _loadingState =
+        MutableStateFlow<Boolean>(true)
+    val loadingState = _loadingState.asStateFlow()
```

Comment:
add addition class or extensions or whatever you want to avoid two variables each time

Reply:
done